app/controllers/application_controller.rb

```diff
@@ -0,0 +1,7 @@
+class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+
+  def render_404
+    raise ActionController::RoutingError.new('Not Found')
```

Comment:
The 404 not found page doesn't show as such on Heroku.  

https://rubyinrails.com/2018/02/26/rails-render-404-not-found-from-controller-action/